### PR TITLE
Bug 1171657 - Schema changes to support password sync.

### PR DESCRIPTION
**Pull request by rnewman**
This branch has the first commit for the synchronizer at the end. Please ignore that.

Note that a few methods in `SQLiteLogins: SyncableLogins` haven't been implemented yet. Those will be fleshed out when they're called by the synchronizer.

The meat of this PR is around tracking `sync_status`, deletion, and mirroring. There's a reasonable amount of automated test coverage.

### CODE REVIEW

**Comment by rnewman** [comment]
lttm

File: Providers/Profile.swift [hunk 1]
```diff
@@ -286,8 +286,28 @@ public class BrowserProfile: Profile {
...
+    lazy var logins: protocol<BrowserLogins, SyncableLogins> = {
+        return SQLiteLogins(db: self.loginsDB)
+    }()
+
+    private lazy var loginsKey: String? = {
+        if AppConstants.IsDebug {
+            return nil
+        }
+
+        let key = "sqlcipher.key.logins.db"
```
Comment (st3fan) [inline Providers/Profile.swift]: Maybe introduce a PrefsKeys that has all of these?

**Comment by fxios-buildasaur** [comment]
Result of integration 1
Integration took 6 minutes and 2 seconds.
2 build errors: build-errors

**Comment by fxios-buildasaur** [comment]
Result of integration 2
Integration took 6 minutes and 49 seconds.
2 build errors: build-errors

**Comment by rnewman** [comment]
This is done and landed.

**Pull request closed without merging**